src/Faker/Provider/fa_IR/PhoneNumber.php

```diff
@@ -39,8 +35,9 @@ class PhoneNumber extends \Faker\Provider\PhoneNumber
         '0937#######',
         '0937#######',
     );
+
     public static function mobileNumber()
     {
-        return static::numerify(static::randomElement(static::$mobileNumberPrefixes));
+        return static::numerify(static::randomElement(static::$formats));
```

Comment:
Does this change really make sense?

Also see 

* https://en.wikipedia.org/wiki/Telephone_numbers_in_Iran#Landlines
* https://en.wikipedia.org/wiki/Telephone_numbers_in_Iran#Mobile_phones

Reply:
Ah I see..

021, 026 and 031 are landline numbers from when called from within Iran. So logically the $formats number prefixes are equally valid. 

However, the "mobileNumber()" method are using "$formats" which contain only landline numbers.

Moving forward, I think what is right is to create another function call "landlineNumber()" to generate landline numbers, both with or without country code. What do you think? Should I go with this approach?

TBH, I'm pretty new here, for now I am just thinking what make sense to me. If my approach wrong, please correct me so I can learn from the mistake.

Hope to hear from you soon.